htdocs/takepos/index.php

```diff
@@ -978,36 +978,39 @@ function WeighingScale(){
 <?php
 
 // TakePOS setup check
-$sql = "SELECT code, libelle FROM ".MAIN_DB_PREFIX."c_paiement";
-$sql .= " WHERE entity IN (".getEntity('c_paiement').")";
-$sql .= " AND active = 1";
-$sql .= " ORDER BY libelle";
+if (isset($_SESSION["takeposterminal"]) && $_SESSION["takeposterminal"]) {
+	$sql = "SELECT code, libelle FROM " . MAIN_DB_PREFIX . "c_paiement";
+	$sql .= " WHERE entity IN (" . getEntity('c_paiement') . ")";
+	$sql .= " AND active = 1";
+	$sql .= " ORDER BY libelle";
+
+	$resql          = $db->query($sql);
+	$paiementsModes = array();
+	if ($resql) {
+		while ( $obj = $db->fetch_object($resql) ) {
+			$paycode = $obj->code;
+			if ($paycode == 'LIQ') {
+				$paycode = 'CASH';
+			}
+			if ($paycode == 'CHQ') {
+				$paycode = 'CHEQUE';
+			}
 
-$resql = $db->query($sql);
-$paiementsModes = array();
-if ($resql) {
-	while ($obj = $db->fetch_object($resql)) {
-		$paycode = $obj->code;
-		if ($paycode == 'LIQ') {
-			$paycode = 'CASH';
-		}
-		if ($paycode == 'CHQ') {
-			$paycode = 'CHEQUE';
+			$constantforkey = "CASHDESK_ID_BANKACCOUNT_" . $paycode . $_SESSION["takeposterminal"];
+			//var_dump($constantforkey.' '.$conf->global->$constantforkey);
+			if ( ! empty($conf->global->$constantforkey) && $conf->global->$constantforkey > 0) {
+				array_push($paiementsModes, $obj);
+			}
 		}
+	}
 
-		$constantforkey = "CASHDESK_ID_BANKACCOUNT_".$paycode.$_SESSION["takeposterminal"];
-		//var_dump($constantforkey.' '.$conf->global->$constantforkey);
-		if (!empty($conf->global->$constantforkey) && $conf->global->$constantforkey > 0) {
-			array_push($paiementsModes, $obj);
-		}
+	if (empty($paiementsModes)) {
```

Comment:
I think this test must be moved outside the parent "if".
So it is still able to report error when $_SESSION["takeposterminal"] is empty, isn't it ?

Reply:
I don't think so. Let me explain.
When you first come to TakePos, $_SESSION["takeposterminal"] isn't set, and you get the terminal selector that, when selected, set the terminal in session. Without that change, an error shows when the terminal selector is shown.